client/src/containers/CampaignAdmin/CampaignAdmin.js

```diff
@@ -387,114 +290,145 @@ By removing this item all linked items will be removed unless used elsewhere.`;
     }
 
     return <Custom key={props.name} {...props} />;
-  }
+  };
 
-  addCampaign() {
-    const path = this.getActionRoute(0, 'create');
-    this.props.router.navigate(path);
-  }
+  const addCampaign = () => {
+    const path = getActionRoute(0, 'create');
+    router.navigate(path);
+  };
+
+  const prevPropsRef = useRef();
+
+  useEffect(() => {
+    // Ensure default breadcrumbs are setup
+    if (breadcrumbs.length === 0) {
+      setBreadcrumbs(router.params.view, router.params.id, title);
+    }
+  }, [breadcrumbs, router.params.view, router.params.id, title, setBreadcrumbs]);
```

Comment:
```suggestion
  useEffect(() => {
    // Ensure default breadcrumbs are setup
    if (breadcrumbs.length === 0) {
      setBreadcrumbs(router.params.view, router.params.id, title);
    }
  }, []);
```

Looks like this is meant to replace `componentDidMount`, but it has a bunch of dependencies? Replacing `componentDidMount()` usually has just an empty array instead. Am I missing some context here?

Reply:
Fixed